sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -530,7 +666,141 @@ public SyncPoller<ShareFileCopyInfo, Void> beginCopy(String sourceUrl, FileSmbPr
+                    throw LOGGER.logExceptionAsError(
+                        new IllegalArgumentException("Cannot cancel a poll response that never started."));
+                }
+                final String copyIdentifier = firstResponse.getValue().getCopyId();
+                if (!CoreUtils.isNullOrEmpty(copyIdentifier)) {
+                    LOGGER.info("Cancelling copy operation for copy id: {}", copyIdentifier);
+                    abortCopyWithResponse(copyIdentifier, finalRequestConditions, null, null);
+                    return firstResponse.getValue();
+                }
+                return null;
+            };
+
+        Function<PollingContext<ShareFileCopyInfo>, Void> fetchResultOperation = (pollingContext) -> null;
+        return SyncPoller.createPoller(interval, syncActivationOperation, pollOperation, cancelOperation, fetchResultOperation);
+    }
+
+    private PollResponse<ShareFileCopyInfo> onPoll(PollResponse<ShareFileCopyInfo> pollResponse,
+                                                   ShareRequestConditions requestConditions) {
+        if (pollResponse.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED
+            || pollResponse.getStatus() == LongRunningOperationStatus.FAILED) {
+            return pollResponse;
+        }
+
+        final ShareFileCopyInfo lastInfo = pollResponse.getValue();
+        if (lastInfo == null) {
+            LOGGER.warning("ShareFileCopyInfo does not exist. Activation operation failed.");
+            return new PollResponse<>(LongRunningOperationStatus.fromString("COPY_START_FAILED", true), null);
+        }
+
+        try {
+            Response<ShareFileProperties> response = getPropertiesWithResponse(requestConditions, null, null);
+            ShareFileProperties value = response.getValue();
+            final CopyStatusType status = value.getCopyStatus();
+            final ShareFileCopyInfo result = new ShareFileCopyInfo(value.getCopySource(), value.getCopyId(),
+                status, value.getETag(), value.getCopyCompletionTime(), value.getCopyStatusDescription());
+
+            LongRunningOperationStatus operationStatus;
+            switch (status) {
+                case SUCCESS:
+                    operationStatus = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                    break;
+                case FAILED:
+                    operationStatus = LongRunningOperationStatus.FAILED;
+                    break;
+                case ABORTED:
+                    operationStatus = LongRunningOperationStatus.USER_CANCELLED;
+                    break;
+                case PENDING:
+                    operationStatus = LongRunningOperationStatus.IN_PROGRESS;
+                    break;
+                default:
+                    throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                        "CopyStatusType is not supported. Status: " + status));
```

Comment:
Let's try and share common logic instead of each client defining it. Especially scenarios like this where it isn't something simple